paasta_tools/frameworks/native_scheduler.py

```diff
@@ -297,8 +286,8 @@ def statusUpdate(self, driver, update):
                 self.tasks_with_flags.pop(task)
 
         if task_params.mesos_task_state is not TASK_STAGING:
-            self.starting_tasks[task_id].cancel()
-            self.starting_tasks = {k: v for k, v in self.starting_tasks if k is not task_id}
+            self.tasks_with_flags[task_id].startup_timer.cancel()
```

Comment:
maybe worth checking that timer still exists? it gets cancelled on statusUpdate about STAGING -> STARTING but then will fail on statusUpdate about RUNNING?